documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java

```diff
@@ -184,19 +186,20 @@ public static List<AbstractNode> toNodes(IdentityManager identityManager,
         LOG.warn("Error getting Folder Node for search result with path {}", node, e);
       }
     }
-    Collections.sort(fileNodes, new Comparator<AbstractNode>(){
-      public int compare(AbstractNode s1, AbstractNode s2) {
-        return s1.getName().compareToIgnoreCase(s2.getName());
-      }
-    });
-    Collections.sort(fileNodes, new Comparator<AbstractNode>(){
-      public int compare(AbstractNode n1, AbstractNode n2) {
-        return Boolean.compare(n2.isFolder(),n1.isFolder());
-      }
-    });
-
-    if(limit < fileNodes.size()){
-      return fileNodes.subList(offset,limit);
+    if (filter.getSortField().equals(DocumentSortField.MODIFIED_DATE)) {
```

Comment:
For the sort and filtering, I think that it would be better to include those conditions inside the query that we will use to request backend (your first proposition was the good one I think https://github.com/exoplatform/documents/compare/677e182b29aed037f72934285acdbf55f355adc5..de05a81ca997b0e4f841b8b4958b296c3acc530b#diff-f5891a2641ecd8cc9f48b7122a7a6fd28c71b2c1a81798457ec6f6c47dcfed2cL258-L263)
Let's enhance this proposition to get a pertinent information from store instead of making the sort manually after retrieving all child nodes of a parent (which can lead to a very bad performances for a parent node having more than 1000 JCR nodes).